plugin/contrib/mesos/pkg/executor/executor.go

```diff
@@ -231,6 +245,26 @@ func (k *KubernetesExecutor) onInitialRegistration() {
 	}
 }
 
+// InitializeStaticPodsSource blocks until initial regstration is complete and
+// then creates a static pod source using the given factory func.
+func (k *KubernetesExecutor) InitializeStaticPodsSource(sourceFactory func()) {
+	<-k.initialRegComplete
+
+	if k.staticPodsConfig == nil {
+		return
+	}
+
+	log.V(2).Infof("extracting static pods config to %s", k.staticPodsConfigPath)
+	err := archive.UnzipDirectory(k.staticPodsConfig, k.staticPodsConfigPath)
+	if err != nil {
+		log.Errorf("Failed to extract static pod config: %v", err)
+		return
+	}
+
+	log.V(2).Infof("initializing static pods source factory, configured at path %q", k.staticPodsConfigPath)
+	sourceFactory()
```

Comment:
Agree, a simple channel would do it as well. But IMO this feels like complicating the code. Maybe we are just too used to callbacks such that they feel natural.